File: backend/apiserver/models/dbmodels/share.go
package dbmodels

import "time"

type Share struct {
	ID            int       `db:"id"`
	UserID        int       `db:"user_id"`
	UpParagraph   string    `db:"up_paragraph"`
	DownParagraph string    `db:"down_paragraph"`
	Sentence      string    `db:"sentence"`
	CreatedAt     time.Time `db:"created_at"`
}

type ShareComment struct {
	ID        int       `db:"id"`
	ShareID   int       `db:"share_id"`
	UserID    int       `db:"user_id"`
	Content   string    `db:"content"`
	Rate      int       `db:"rate"`
	CreatedAt time.Time `db:"created_at"`
}

File: frontend/App/Models/Share.swift
//
// Copyright © 2019 Sunho Kim. All rights reserved.
//

import Foundation

class Share: Codable {
    var id: Int = 0
    var userId: Int = 0
    var upParagraph: String = ""
    var downParagraph: String = ""
    var sentence: String = ""
    var createdAt: String = ""
}

class SharedCommend: Codable {
    var id: Int = 0
    var shareId: Int = 0
    var userId: Int = 0
    var content: String = ""
    var rate: Int = 0
}
